api/src/main/java/jakarta/data/metamodel/TextAttribute.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023,2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+import jakarta.data.Sort;
+
+/**
+ * Represents an textual entity attribute in the {@link StaticMetamodel}.
+ */
+public interface TextAttribute extends SortableAttribute {
```

Comment:
I would remove this interface and keep the static metamodel as simple as possible.

Reply:
> I would remove this interface and keep the static metamodel as simple as possible.

If we consolidate it all onto SortableAttribute, then anything that is a SortableAttribute will be able to request  .ascIgnoreCase() and .descIgnoreCase() which means letting users write nonsense code for numeric fields.  By having these specific types, it makes a hierarchy of interfaces that control what you can actually do with the attributes based on what types they are.  That is a superior model because it stops errors from being introduced while code is being developed.